diff -ru ngx_openresty-1.9.3.1.orig/bundle/nginx-1.9.3/src/event/ngx_event_openssl.c ngx_openresty-1.9.3.1/bundle/nginx-1.9.3/src/event/ngx_event_openssl.c
--- ngx_openresty-1.9.3.1.orig/bundle/nginx-1.9.3/src/event/ngx_event_openssl.c	2015-08-12 08:23:31.836906397 -0300
+++ ngx_openresty-1.9.3.1/bundle/nginx-1.9.3/src/event/ngx_event_openssl.c	2015-08-12 08:24:16.162290810 -0300
@@ -2942,7 +2942,7 @@
         goto failed;
     }
 
-    if (X509_check_host(cert, (char *) name->data, name->len, 0, NULL) != 1) {
+    if (X509_check_host(cert, (char *) name->data, name->len, X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT, NULL) != 1) {
         ngx_log_debug0(NGX_LOG_DEBUG_EVENT, c->log, 0,
                        "X509_check_host(): no match");
         goto failed;
@@ -2997,7 +2997,10 @@
                        "SSL subjectAltName: no match");
 
         GENERAL_NAMES_free(altnames);
-        goto failed;
+        /* Line below is commented in order to reproduce
+         * X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT behavior
+         */
+        //goto failed;
     }
 
     /*
Somente em ngx_openresty-1.9.3.1/bundle/nginx-1.9.3/src/event: ngx_event_openssl.c.orig
