diff -ru openresty-1.9.7.3.orig/bundle/nginx-1.9.7/src/event/ngx_event_openssl.c openresty-1.9.7.3/bundle/nginx-1.9.7/src/event/ngx_event_openssl.c
--- openresty-1.9.7.3.orig/bundle/nginx-1.9.7/src/event/ngx_event_openssl.c	2016-01-29 08:48:16.797204117 -0200
+++ openresty-1.9.7.3/bundle/nginx-1.9.7/src/event/ngx_event_openssl.c	2016-01-29 08:48:44.734074858 -0200
@@ -2956,7 +2956,7 @@
         goto failed;
     }
 
-    if (X509_check_host(cert, (char *) name->data, name->len, 0, NULL) != 1) {
+    if (X509_check_host(cert, (char *) name->data, name->len, X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT, NULL) != 1) {
         ngx_log_debug0(NGX_LOG_DEBUG_EVENT, c->log, 0,
                        "X509_check_host(): no match");
         goto failed;
@@ -3011,7 +3011,10 @@
                        "SSL subjectAltName: no match");
 
         GENERAL_NAMES_free(altnames);
-        goto failed;
+        /* Line below is commented in order to reproduce
+         * X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT behavior
+         */
+        //goto failed;
     }
 
     /*
